### PR TITLE
Slim 739 align sss smart meter integration test suite

### PR DESCRIPTION
**Pull request by robinbakkeruscgi**

### CODE REVIEW

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/platform/dlms/cucumber/steps/ws/smartmetering/smartmeteringmanagement/GetDebugInformation.java [hunk 1]
```diff
@@ -64,8 +64,10 @@ public void theGetDebugInformationRequestIsReceived(final Map<String, String> re
     public void theDeviceDebugInformationShouldBeInTheResponseMessage() throws Throwable {
         PROPERTIES_MAP.put(Keys.KEY_DEVICE_IDENTIFICATION, ScenarioContext.Current()
                 .get(Keys.KEY_DEVICE_IDENTIFICATION).toString());
-        PROPERTIES_MAP
-        .put(Keys.KEY_CORRELATION_UID, ScenarioContext.Current().get(Keys.KEY_CORRELATION_UID).toString());
+
+        // PROPERTIES_MAP
+        // .put(Keys.KEY_CORRELATION_UID,
+        // ScenarioContext.Current().get(Keys.KEY_CORRELATION_UID).toString());
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/platform/dlms/cucumber/steps/ws/smartmetering/smartmeteringmanagement/GetDebugInformation.java]: remove these lines

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.text.ParseException;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.alliander.osgp.platform.cucumber.helpers.UtcDateHelper;
+
+public class TestUTCDate {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: Rename the class to match the java conventions

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    @Test
+    public void test() throws ParseException {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: add a comment to clarify what you are testing

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        final Date utc = UtcDateHelper.getUtcDate();
+        Assert.assertTrue("utc date is before actual date", utc.before(new Date()));
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: depends on the timezone where the code is run if the test will pass or fail

File: cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringAdhoc.feature [hunk 5]
```diff
@@ -8,6 +8,7 @@ Feature: SmartMetering AdHoc
       | DeviceType           | SMART_METER_E     |
       | Active               | true              |
 
+@TEST
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringAdhoc.feature]: remove this tag

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/core/builders/FirmwareBuilder.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+        if (inputSettings.containsKey(Keys.FIRMWARE_FILE)) {
+            this.withFile(this.getBytesFormFile(inputSettings));
+        }
+
+        return this;
+    }
+
+    private byte[] getBytesFormFile(final Map<String, String> inputSettings) {
+        // TODO
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/core/builders/FirmwareBuilder.java]: remove the empty method or code what needs to be coded

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 7]
```diff
@@ -78,4 +78,30 @@
     public static final String KEY_MEASUREMENT_VALUE = "MeasurementValue";
 
     public static final String KEY_RESULT = "Result";
+
+    public static final String FIRMWARE_FILENAME = "FirmwareFilename";
+    public static final String FIRMWARE_DESCRIPTION = "FirmwareDescription";
+    public static final String FIRMWARE_PUSHTONEWDEVICES = "FirmwarePushToNewDevices";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_PUSHTONEWDEVICES ->FIRMWARE_PUSH_TO_NEW_DEVICES

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 8]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_MODULEVERSIONCOMM = "FirmwareModuleVersionComm";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_MODULEVERSIONCOMM -> FIRMWARE_MODULE_VERSION_COMM

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 9]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_MODULEVERSIONFUNC = "FirmwareModuleVersionFunc";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_MODULEVERSIONFUNC -> FIRMWARE_MODULE_VERSION_FUNC

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 10]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_MODULEVERSIONMA = "FirmwareModuleVersionMa";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_MODULEVERSIONMA -> FIRMWARE_MODULE_VERSION_MA

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 11]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_MODULEVERSIONMBUS = "FirmwareModuleVersionMbus";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_MODULEVERSIONMBUS -> FIRMWARE_MODULE_VERSION_MBUS

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 12]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_MODULEVERSIONSEC = "FirmwareModuleVersionSec";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: FIRMWARE_MODULEVERSIONSEC -> FIRMWARE_MODULE_VERSION_SEC

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 13]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String FIRMWARE_FILE = "FirmwareFile";
+    public static final String FIRMWARE_HASH = "FirmwareHash";
+    public static final String FIRMWARE_VERSION = "FirmwareVersion";
+
+    public static final String MANUFACTURER_ID = "ManufacturerId";
+    public static final String MANUFACTURER_NAME = "ManufacturerName";
+    public static final String MANUFACTURER_USEPREFIX = "ManufacturerUsePrefix";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: MANUFACTURER_USEPREFIX -> MANUFACTURER_USE_PREFIX

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 14]
```diff
@@ -78,4 +78,30 @@
...
+
+    public static final String DEVICEFIRMWARE_INSTALLATIONDATE = "DevicefirmwareInstallationDate";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: DEVICEFIRMWARE_INSTALLATIONDATE -> DEVICEFIRMWARE_INSTALLATION_DATE

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 15]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String DEVICEFIRMWARE_INSTALLEDBY = "DevicefirmwareInstalledBy";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: DEVICEFIRMWARE_INSTALLEDBY -> DEVICEFIRMWARE_INSTALLED_BY

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java [hunk 16]
```diff
@@ -78,4 +78,30 @@
...
+    public static final String DEVICEFIRMWARE_ACTIVE = "DevicefirmwareActive";
+
+    public static final String DEVICEMODEL_MODELCOE = "DeviceModelmodelCode";
+    public static final String DEVICEMODEL_DESCRIPTION_ = "DeviceModelDescription";
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: DEVICEMODEL_DESCRIPTION_ -> DEVICEMODEL_DESCRIPTION

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/platform/dlms/cucumber/steps/ws/smartmetering/smartmeteringmanagement/GetDebugInformation.java (hunk 1 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/platform/dlms/cucumber/steps/ws/smartmetering/smartmeteringmanagement/GetDebugInformation.java]: done

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java (hunk 2 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: done

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java (hunk 3 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: done

Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUTCDate.java]: done

File: cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringAdhoc.feature (hunk 5 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringAdhoc.feature]: done

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/core/builders/FirmwareBuilder.java (hunk 6 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/core/builders/FirmwareBuilder.java]: removed this code

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java (hunk 16 above)
Comment (robinbakkeruscgi) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/platform/cucumber/steps/Keys.java]: all done

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/MicrogridsGetDataPv.feature [hunk 17]
```diff
@@ -1,9 +1,9 @@
-@Iec61850MockServer
 Feature: Get PhotoVoltaic System Data
   In order to be able to know data of a photovoltaic system with a remote terminal unit
   As an OSGP client
   I want to get PV data from an RTU
 
+@Iec61850MockServer
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/MicrogridsGetDataPv.feature]: Iec61850MockServer is needed for the whole feature. Revert this change.

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUtcDateHelper.java [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.alliander.osgp.platform.cucumber.helpers.UtcDateHelper;
+
+public class TestUtcDateHelper {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/test/TestUtcDateHelper.java]: remove this test

File: cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/MicrogridsGetDataPv.feature [hunk 19]
```diff
@@ -3,7 +3,6 @@ Feature: Get PhotoVoltaic System Data
   As an OSGP client
   I want to get PV data from an RTU
 
-@Iec61850MockServer
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/resources/features/osgp-adapter-ws-microgrids/MicrogridsGetDataPv.feature]: tag should be on the feature
